### ISSUE TITLE
Prices 

### ISSUE THREAD

**Issue by tux-rampage** [magento/magento2#35281]
### Preconditions (*)
Magento version: 2.4.3-p1
Applied Patches:
 - MDVA-43395
 - MDVA-43443

1. Define a store Locale like de_DE
2. Define a **different** store currency than the locale's default (like CHF)

### Steps to reproduce (*)

1. Add products with actual prices to the catalog
2. Clear all caches
3. View a product in the store frontend
4. Add the product to the cart

### Expected result (*)
1. The product should show the correct price in pricebox
2. Cart totals should show correct calculated totals

### Actual result (*)

1. Every product shows a zero zrice
3. All totals show zero prices

### Trackdown

Origin is the implementation in `Magento\Directory\Model\Currency`:

```php
    /**
     * @return string
     */
    public function getOutputFormat()
    {
        $formatted = $this->formatTxt(0);
        $number = $this->formatTxt(0, ['display' => \Magento\Framework\Currency::NO_SYMBOL]);
        return str_replace($this->trimUnicodeDirectionMark($number), '%s', $formatted);
    }
```

In this segment `$number` is expected to be the formatted zero number without currency symbol. This is not the case because of a bug in `formatCurrency()` which is used by `formatTxt()`.

```php
    /**
     * Format currency.
     *
     * @param string $price
     * @param array $options
     * @return string
     */
    private function formatCurrency(string $price, array $options): string
    {
        $customerOptions = new \Magento\Framework\DataObject([]);

        $this->_eventManager->dispatch(
            'currency_display_options_forming',
            ['currency_options' => $customerOptions, 'base_code' => $this->getCode()]
        );
        $options += $customerOptions->toArray();

        $this->numberFormatter = $this->getNumberFormatter($options);

        $formattedCurrency = $this->numberFormatter->formatCurrency(
            $price, $this->getCode() ?? $this->numberFormatter->getTextAttribute(\NumberFormatter::CURRENCY_CODE)
        );

        if (array_key_exists(LocaleCurrency::CURRENCY_OPTION_SYMBOL, $options)) {
            // remove only one non-breaking space from custom currency symbol to allow custom NBSP in currency symbol
            $formattedCurrency = preg_replace('/ /u', '', $formattedCurrency, 1);
        }

        if ((array_key_exists(LocaleCurrency::CURRENCY_OPTION_DISPLAY, $options)
                && $options[LocaleCurrency::CURRENCY_OPTION_DISPLAY] === \Magento\Framework\Currency::NO_SYMBOL)) {
            $formattedCurrency = str_replace(' ', '', $formattedCurrency);
        }

        return preg_replace('/^\s+|\s+$/u', '', $formattedCurrency);
    }
```

This method uses PHP intl's `NumberFormatter`. This will **always** add a currency symbol when `formatCurrency()` is called with a different currency code than the locale's default one (Since `NumberFormatter::setSymbol(NumberFormatter::CURRENCY_SYMBOL)` only refers to the locale's currency). Therefore the rest of the code breaks because of incorrect assumptions.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [x] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by tux-rampage** [comment]
This may be fixed with Commit 57a9adfd85dfe1a1235d91380f8d7580d5c49ec0

**Comment by tux-rampage** [comment]
It seems also be dependant on the system's `intl` library:

* php 7.4 on alpine (Docker-Image) is affected
* php 7.4 on Ubuntu 20.04 is not affected

**Comment by engcom-November** [comment]
Verified the issue on Magento 2.4-develop branch but the issue is not reproducible:
@tux-rampage  ,
This issue is a duplicate of https://github.com/magento/magento2/issues/33798. Similar https://github.com/magento/magento2/issues/33856 was fixed in the 2.4-develop branch in [ae304d4,](https://github.com/magento/magento2/commit/ae304d43cd36322b1ff8145d70c9b2e9aeb0cde) As a workaround, you could apply it as patch.

https://github.com/magento/magento2/issues/33856#issuecomment-904606945
Kindly recheck the issue on Magento 2.4-develop branch and provide missing steps if any if the issue is still reproducible.
Thank you.

**Comment by engcom-November** [comment]
Hi @tux-rampage ,
We are closing this issue as there has been no latest update on the same.
Thank you.
